### PR TITLE
executor: show more information about cop tasks in slow log

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Show more information about cop tasks in slow log.

### What is changed and how it works?
1. Add `allExecDetails` in `StatementContext` to record all cop-tasks' execution information.
2. Add `CopTasksDetails` to calculate and generate some useful information about cop-tasks.
3. Update `SlowLogFormat` to put cop-task execution information into slow log.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: sessionctx/stmtctx/stmtctx.go [hunk 1]
```diff
@@ -95,6 +97,7 @@ type StatementContext struct {
 		warnings          []SQLWarn
 		histogramsNotLoad bool
 		execDetails       execdetails.ExecDetails
+		allExecDetails    []*execdetails.ExecDetails
```
Comment (lamxTyler) [inline sessionctx/stmtctx/stmtctx.go]: Should we reset it in `ResetForRetry`?

Comment (qw4990) [reply sessionctx/stmtctx/stmtctx.go]: Yes... you are right.
This problem has been fixed.

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -957,6 +961,9 @@ func (s *SessionVars) SlowLogFormat(txnTS uint64, costTime time.Duration, execDe
 		}
 		buf.WriteString("\n")
 	}
+	if copTasks != nil {
+		buf.WriteString(SlowLogRowPrefixStr + SlowLogCopTasks + SlowLogSpaceMarkStr + copTasks.String() + "\n")
```
Comment (zz-jason) [inline sessionctx/variable/session.go]: this line is too long, can we split it to three lines:
```
Num_cop_tasks: %v
Cop_process: Avg_time: %v P90_time: %v Max_time: %v Max_addr: %v
Cop_wait: Avg_time: %v P90_time: %v Max_time: %v Max_Addr: %v
```

**Pull request merged**